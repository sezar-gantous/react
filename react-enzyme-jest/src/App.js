import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <ul>
          <li>Test 1</li>
          <li>Test 2</li>
          <li>Test 3</li>
        </ul>
        <p className="App-intro">Hello World</p>
      </div>
    );
  }
}

export class Link extends Component {
  render() {
    return <a href={this.props.address}>Click</a>;
  }
}

export default App;
