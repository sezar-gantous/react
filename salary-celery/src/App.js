import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Main from "./components/main";
import MenuBar from "./components/MenuBar";

const styles = theme => ({
  root: {
    flexGrow: 1,
    paddingLeft: theme.spacing.unit,
    paddingRight: theme.spacing.unit
  }
});

class App extends Component {
  render() {
    const { classes } = this.props;

    return (
      <Grid container className={cx(App, classes.root)} justify="center">
        <Grid item xs={12} md={6} lg={4}>
          <Grid container spacing={16} justify="center">
            <Grid item xs={12}>
              <MenuBar />
            </Grid>
            <Grid item xs={12}>
              <Main />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    );
  }
}

App.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(App);
