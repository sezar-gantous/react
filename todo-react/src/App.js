import React, { Component } from "react";
import PropTypes from "prop-types";
import logo from "./logo.svg";
import "./App.css";
import { TodoForm, TodoList, Footer } from "./components/todo";
import {
  addTodo,
  generateId,
  findById,
  toggleTodo,
  updateTodo,
  removeTodo,
  filterTodos
} from "./lib/todoHelpers";
import { pipe, partial } from "./lib/utils";
import {
  loadTodos,
  createTodo,
  saveTodo,
  destroyTodo
} from "./lib/todoService";

class App extends Component {
  state = {
    todos: [],
    currentTodo: ""
  };

  static contextTypes = {
    route: PropTypes.string
  };

  componentDidMount() {
    loadTodos()
      .then(todos => this.setState({ todos }))
      .catch(err => this.setState({ errorMsg: `${err}` }));
  }

  handleRemove = (id, evt) => {
    evt.preventDefault();
    const updateTodos = removeTodo(this.state.todos, id);
    this.setState({ todos: updateTodos });
    destroyTodo(id)
      .then(() => this.showTempMessage("Todo Removed"))
      .catch(err => this.setState({ errorMsg: `${err}` }));
  };

  handleToggle = id => {
    const getToggleTodo = pipe(
      findById,
      toggleTodo
    );
    const updated = getToggleTodo(id, this.state.todos);
    const getUpdatedTodos = partial(updateTodo, this.state.todos);

    const updatedTodos = getUpdatedTodos(updated);

    this.setState({ todos: updatedTodos });
    saveTodo(updated)
      .then(() => this.showTempMessage("Todo Upaded"))
      .catch(err => this.setState({ errorMsg: `${err}` }));
  };

  handleSubmit = e => {
    e.preventDefault();

    const newId = generateId();
    const newTodo = {
      name: this.state.currentTodo,
      isComplete: false,
      id: newId
    };

    const updatedTodo = addTodo(this.state.todos, newTodo);

    this.setState({
      todos: updatedTodo,
      currentTodo: "",
      errorMsg: ""
    });

    createTodo(newTodo)
      .then(() => this.showTempMessage("Todo added"))
      .catch(err => this.setState({ errorMsg: `${err}` }));
  };

  showTempMessage = msg => {
    this.setState({ message: msg });
    setTimeout(() => this.setState({ message: "" }), 2500);
  };

  handleEmptySubmit = e => {
    e.preventDefault();
    this.setState({
      errorMsg: "Please supply a todo name"
    });
  };

  handleInputChange = e => {
    this.setState({
      currentTodo: e.target.value
    });
  };
  render() {
    const submitHandler = this.state.currentTodo
      ? this.handleSubmit
      : this.handleEmptySubmit;
    const displayTodos = filterTodos(this.state.todos, this.context.route);
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>React Todos</h2>
        </div>
        <div className="Todo-App">
          {this.state.errorMsg && (
            <span className="error"> {this.state.errorMsg}</span>
          )}

          {this.state.message && (
            <span className="success"> {this.state.message}</span>
          )}
          <TodoForm
            handleInputChange={this.handleInputChange}
            currentTodo={this.state.currentTodo}
            handleSubmit={submitHandler}
          />
          <TodoList
            handleToggle={this.handleToggle}
            todos={displayTodos}
            handleRemove={this.handleRemove}
          />

          <Footer />
        </div>
      </div>
    );
  }
}

export default App;
