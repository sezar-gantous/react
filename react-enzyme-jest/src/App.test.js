import React from "react";
import ReactDOM from "react-dom";
import App, { Link } from "./App";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<App/>", () => {
  const wrapper = shallow(<App />);
  it("h1 contains correct text", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("h1").text()).toBe("Welcome to React");
  });

  it("matches the snapshot", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});

describe("<Link/>", () => {
  it("link component accepts address prop", () => {
    const warraper = shallow(<Link address="www.google.com" />);
    expect(warraper.instance().props.address).toBe("www.google.com");
  });

  it("a tag node renders href correctly", () => {
    const warraper = shallow(<Link address="www.google.com" />);
    expect(warraper.props().href).toBe("www.google.com");
  });
});
