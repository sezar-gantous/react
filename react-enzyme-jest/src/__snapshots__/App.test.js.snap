// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
      <h3
        className="blue"
      >
        Everyone is Welcome!
      </h3>
    </header>
    <Form>
      <form
        data-testid="addUserForm"
        onSubmit={[Function]}
      >
        <h2>
          Request Information
        </h2>
        <input
          data-testid="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
        />
        <input
          data-testid="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
        <input
          data-testid="number"
          onChange={[Function]}
          placeholder="Number"
          type="text"
          value=""
        />
        <div>
          <input
            checked={true}
            data-testid="checked"
            onChange={[Function]}
            onClick={[Function]}
            type="checkbox"
          />
          <p
            className="promotions"
            data-testid="promotionsP"
          >
            Receive Promotions
          </p>
        </div>
        <button
          data-testid="submitButton"
          type="submit"
        >
          Submit
        </button>
      </form>
    </Form>
    <p
      className="App-intro"
    >
      Hello World
    </p>
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
    <p
      className="lifeCycle"
    >
      componentDidMount
    </p>
    *
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h3
      className="blue"
    >
      Everyone is Welcome!
    </h3>
  </header>
  <Form />
  <p
    className="App-intro"
  >
    Hello World
  </p>
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
  <p
    className="lifeCycle"
  >
    componentDidMount
  </p>
  *
</div>
`;
